<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.8.26">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>lab_molec_edna_qpcr – Gatins Lab Protocols</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js" type="module"></script>
<script src="site_libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="site_libs/quarto-html/axe/axe-check.js" type="module"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-587c61ba64f3a5504c4d52d930310e48.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap-d0dac7337b49546037d11b12c2873d1a.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


<link rel="stylesheet" href="styles.css">
</head>

<body class="nav-sidebar docked nav-fixed quarto-light">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="./index.html">
    <span class="navbar-title">Gatins Lab Protocols</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="./index.html"> 
<span class="menu-text">Welcome</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link active" href="./lab_generalprotocols.html" aria-current="page"> 
<span class="menu-text">Lab Protocols</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./fieldprotocols.html"> 
<span class="menu-text">Field Protocols</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./prof_development.html"> 
<span class="menu-text">Professional Development</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./coding.html"> 
<span class="menu-text">Coding</span></a>
  </li>  
</ul>
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/remygatins"> <i class="bi bi-github" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://twitter.com/remygatins?lang=en"> <i class="bi bi-twitter" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
</div>
      </div> <!-- /container-fluid -->
    </nav>
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./lab_molec_etoh.html">Molecular Protocols</a></li><li class="breadcrumb-item"><a href="./lab_molec_eDNA_filtration.html">eDNA Protocols</a></li><li class="breadcrumb-item"><a href="./lab_molec_eDNA_qpcr.html">eDNA qPCR Protocol (BSB)</a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation docked overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header sidebar-header-stacked">
      <a href="./index.html" class="sidebar-logo-link">
      </a>
      </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="false">
 <span class="menu-text">General Protocols</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./lab_generalprotocols.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">General Protocols</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./lab_gen_sharedlabspace.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">NUMSC - Shared Lab Space</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./lab_gen_pipetting.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Pipetting</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./lab_gen_water.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Types of Water</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./lab_gen_stableisotope.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Stable Isotope Protocol</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="false">
 <span class="menu-text">Equipment</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./lab_equip_80_freezer.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">-80°C Freezer</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./lab_equip_autoclave.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Autoclave (OGL)</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./lab_equip_tapestation.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Tapestation</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./lab_equip_qubit.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">QubitFlex Fluorometer</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./lab_equip_ph_meter.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">pH Meter</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="true">
 <span class="menu-text">Molecular Protocols</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./lab_molec_etoh.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Types of Ethanol</span></a>
  </div>
</li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="true">
 <span class="menu-text">eDNA Protocols</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-4" class="collapse list-unstyled sidebar-section depth2 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./lab_molec_eDNA_filtration.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">eDNA Water Collection &amp; Filtration</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./lab_molec_eDNA_extraction.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">eDNA Filter Extraction</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./lab_molec_eDNA_qpcr.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">eDNA qPCR Protocol (BSB)</span></a>
  </div>
</li>
      </ul>
  </li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./lab_molec_dnaextraction.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Qiagen DNeasy extraction</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./lab_molec_DNA_beadcleanup.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">DNA Bead Cleanup</span></a>
  </div>
</li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" role="navigation" aria-expanded="false">
 <span class="menu-text">Illumina WGS Library Prep</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-5" class="collapse list-unstyled sidebar-section depth2 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./lab_molec_illuminalibraryprep.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Illumina Library Prep</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./lab_molec_yinlineannealing.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Annealing Y-in line adapters for use in Illumina WGS</span></a>
  </div>
</li>
      </ul>
  </li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./lab_molec_gel_electrophoresis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Gel Electrophoresis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./lab_molec_low_te_buffer.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Low TE Buffer</span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#edna-qpcr-protocol-bsb" id="toc-edna-qpcr-protocol-bsb" class="nav-link active" data-scroll-target="#edna-qpcr-protocol-bsb">eDNA qPCR Protocol (BSB)</a>
  <ul class="collapse">
  <li><a href="#qpcr-profile" id="toc-qpcr-profile" class="nav-link" data-scroll-target="#qpcr-profile">qPCR profile</a></li>
  <li><a href="#gblock-plate-and-standard-curve" id="toc-gblock-plate-and-standard-curve" class="nav-link" data-scroll-target="#gblock-plate-and-standard-curve">gblock plate and standard curve</a></li>
  <li><a href="#qpcr-for-samples" id="toc-qpcr-for-samples" class="nav-link" data-scroll-target="#qpcr-for-samples">qPCR for samples</a></li>
  <li><a href="#calculating-copy-number" id="toc-calculating-copy-number" class="nav-link" data-scroll-target="#calculating-copy-number">Calculating Copy Number</a></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content"><header id="title-block-header" class="quarto-title-block"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./lab_molec_etoh.html">Molecular Protocols</a></li><li class="breadcrumb-item"><a href="./lab_molec_eDNA_filtration.html">eDNA Protocols</a></li><li class="breadcrumb-item"><a href="./lab_molec_eDNA_qpcr.html">eDNA qPCR Protocol (BSB)</a></li></ol></nav></header>




<section id="edna-qpcr-protocol-bsb" class="level1">
<h1>eDNA qPCR Protocol (BSB)</h1>
<p><a href="https://github.com/remygatins/GatinsLabProtocols/edit/main/lab_molec_eDNA_qpcr.qmd">Edit this page</a></p>
<section id="qpcr-profile" class="level2">
<h2 class="anchored" data-anchor-id="qpcr-profile">qPCR profile</h2>
<p><img width="430" alt="" src="pictures/eDNAqPCR_profile.png"> <img width="552" alt="" src="pictures/eDNAqPCR_profile2.png"></p>
<p><strong>RESOURCES:</strong></p>
<ul>
<li><a href="https://docs.google.com/spreadsheets/d/1mhRRfssO4oSUR1PfabGZuQd-7WKd-uLb7ejLS4Qbwyo/edit?usp=sharing">BSB eDNA qPCR supplies</a></li>
<li><a href="https://docs.google.com/document/d/1mVxfi3i9k1xxNkycKfKhOgId1RCR28_PBVNqh4CJs2Y/edit?usp=sharing">eDNA BSB qPCR Assay document and protocol resources</a></li>
<li><a href="https://www.br.io">BR.io</a> -&gt; website that qPCR machine is connected to
<ul>
<li>only one BR.io account can be connected to a user at a time</li>
</ul></li>
<li>qPCR machine = Bio-Rad CFX Opus 96 (Real-Time PCR System)
<ul>
<li>username: gatins; password: gatins</li>
<li><a href="https://info.bio-rad.com/CFX-Resource-Hub.html">resource hub for machine</a></li>
</ul></li>
<li>[gBlock dilution calculator &amp; gBlock plate plans]link(https://docs.google.com/spreadsheets/d/1ZA_yLsmvcSCtYSLhoJNwShIfMoxdycmk1gIEXeU-3TY/edit?usp=sharing)</li>
<li><a href="https://docs.google.com/spreadsheets/d/1VQfYPV60tebJGguQfghsxpS0YRRm-WdmMyxA82zIlHk/edit?usp=sharing">sample plate plans</a></li>
</ul>
</section>
<section id="gblock-plate-and-standard-curve" class="level2">
<h2 class="anchored" data-anchor-id="gblock-plate-and-standard-curve">gblock plate and standard curve</h2>
<p><strong>NEED:</strong></p>
<ul class="task-list">
<li><label><input type="checkbox">primer master mix</label>
<ul class="task-list">
<li><label><input type="checkbox">6.91 μl BSB-F</label></li>
<li><label><input type="checkbox">6.91 μl BSB-R</label></li>
<li><label><input type="checkbox">3.46 μl BSB-P</label></li>
<li><label><input type="checkbox">904.32 μl H20</label></li>
</ul></li>
<li><label><input type="checkbox"><a href="https://www.thermofisher.com/order/catalog/product/4396838">TaqMan™ Environmental Master Mix 2.0</a></label></li>
<li><label><input type="checkbox">500 base pair gBlock (iDT; ASK REMY)</label></li>
<li><label><input type="checkbox">ice</label></li>
<li><label><input type="checkbox">15 ml tube</label></li>
<li><label><input type="checkbox">1 or 2 ml microcentrifuge tube</label></li>
<li><label><input type="checkbox">PCR strip tubes</label></li>
<li><label><input type="checkbox">Hard-Shell PCR plate, 96-well, thin wall (<a href="https://www.bio-rad.com/en-us/sku/HSP9601-hard-shell-96-well-pcr-plates-low-profile-thin-wall-skirted-white-clear?ID=HSP9601">Bio-rad #HSP9601</a>; pkg of 50, white shell/clear well)</label></li>
<li><label><input type="checkbox">plate seal/cover (<a href="https://www.bio-rad.com/en-us/sku/17010701-microseal-b-pcr-plate-sealing-film-adhesive-optical?ID=17010701">Bio-rad Microseal ‘B’ seal Seals #MSB1001</a>)</label></li>
</ul>
<p><a href="https://docs.google.com/spreadsheets/d/1ZA_yLsmvcSCtYSLhoJNwShIfMoxdycmk1gIEXeU-3TY/edit?usp=sharing"><strong>PLATE PLAN:</strong></a></p>
<p><img width="900" alt="" src="pictures/eDNAqPCR_gblockplate.png"></p>
<p><strong>PREP:</strong></p>
<ul class="task-list">
<li><label><input type="checkbox">1. wipe down PCR hood and counter with 10% bleach</label></li>
<li><label><input type="checkbox">2. sterilize hood with UV light</label></li>
<li><label><input type="checkbox">3. thaw primer master mix or individual primers/probe (if you have to make more master mix) in ice container with lid to block light (to protect probe)</label>
<ul class="task-list">
<li><label><input type="checkbox">3.1 dilute primers to 300 nM and dilute probe to 150 nM with low TE buffer (pH 8) if they are not yet resuspended</label></li>
</ul></li>
<li><label><input type="checkbox">4. make and print plate plan</label></li>
<li><label><input type="checkbox">5. tape plate plan to hood</label></li>
<li><label><input type="checkbox">6. grab ice</label></li>
</ul>
<p><strong>STEPS:</strong></p>
<ul class="task-list">
<li><p><label><input type="checkbox">1. resuspend gblock if needed: <a href="https://www.idtdna.com/pages/education/decoded/article/tips-for-working-with-gblocks-gene-fragments">IDT protocol</a></label></p>
<ul class="task-list">
<li><label><input type="checkbox">1.1 after resuspending gBlock, store it in aliquots at -20 C to avoid freeze-thawing</label></li>
</ul></li>
<li><p><label><input type="checkbox">2. make gblock concentrations using Tim’s dilution calculator and our dilution calculator (C = 10 or 1 depending on starting concentration; <a href="https://docs.google.com/spreadsheets/d/1ZA_yLsmvcSCtYSLhoJNwShIfMoxdycmk1gIEXeU-3TY/edit?usp=sharing">dilution calculator</a>)</label></p>
<ul class="task-list">
<li><label><input type="checkbox">2.1 start the serial dilution at 10 ng/μl or 1 ng/μl depending on what gBlock aliquots are available</label></li>
<li><label><input type="checkbox">2.2 if starting at 10 ng/μl concentration (which has a copy number of 1.95 x 10^10 ng/μl (copy number/μl = C ∗ M ∗ (1 x 10^-15 mol/fmol) ∗ Avogadro’s number)), dilute with 0.9511 μl of nuclease-free water to reach a copy number of 10^10 ng/μl</label></li>
<li><label><input type="checkbox">2.3 continue to dilute the gBlock until you have all dilutions needed for the plate (10^8, 10^6, 10^5, 10^4, 10^3, 500, 250, 100, 50, 25, 10, 5, 3, and 1 ng/μl)</label></li>
</ul></li>
<li><p><label><input type="checkbox">3. keep dilutions on ice</label></p></li>
<li><p><label><input type="checkbox">4. make or thaw primer master mix (enough for 96 wells * 1.2 for pipet error)</label></p>
<ul class="task-list">
<li><label><input type="checkbox">4.1 primer master mix = 6.91 μl BSB-F, 6.91 μl BSB-R, 3.46 μl BSB-P, 904.32 μl H20 (keep covered when not using to protect the probe)</label></li>
</ul></li>
<li><p><label><input type="checkbox">5. pipet 1,152 μl of the taquman master mix into a 15 ml tube</label></p></li>
<li><p><label><input type="checkbox">6. then pipet 921.6 μl of the primer master mix into the same tube</label></p></li>
<li><p><label><input type="checkbox">7. swirl gently, and transfer half of the liquid to a 1.5 or 2 ml tube to make it easier to pipet from (transfer the remaining combined master mix once it runs low)</label></p></li>
<li><p><label><input type="checkbox">8. pipet 129.6 μl of the combined master mix into each well of a strip tube</label></p></li>
<li><p><label><input type="checkbox">9. using a multichannel pipet, pipet out 18 μl of strip tube solution into each well of the plate</label></p></li>
<li><p><label><input type="checkbox">10. after 6 columns, pipet out another 129.6 μl of the combined master mix into each well of the same strip tube (might need to pipet it around if running low in specific tubes of the strip tube)</label></p></li>
<li><p><label><input type="checkbox">11. Finish pipetting 18 μl of the strip tube solution into the remaining wells of the plate</label></p></li>
<li><p><label><input type="checkbox">12. working backwards, pipet 2 μl of each dilution into their designated well shown in the plate plan above (work from the nuclease-free water negative control up to 10^8 concentration)</label></p></li>
<li><p><label><input type="checkbox">13. place down plate seal, making sure it covers every well</label></p></li>
<li><p><label><input type="checkbox">14. vortex, then bring to the shared lab space to centrifuge for ~10-20 secounds by holding down the short spin button (you can find a plate balance in the lab next to the empty tube racks)</label></p></li>
<li><p><label><input type="checkbox">15. wipe down seal with a kimwipe</label></p></li>
<li><p><label><input type="checkbox">16. place into Bio-rad qPCR machine and start BSB eDNA run (labeled: BSB-eDNA) which can be found in “Files” under “My Files”</label></p>
<ul>
<li>BSB-eDNA Protocol File:
<ul>
<li>Modified: Aug 28, 2024, 2:13 PM</li>
<li>Method: CALC</li>
<li>Lid Temperature: 105 °C</li>
<li>Reaction Volume: 20 μl
<ul>
<li>Steps:
<ol type="1">
<li>95 °C, 00:10:00</li>
<li>95 °C, 00:00:10</li>
<li>60 °C, 00:00:45 Plate Read</li>
<li>Goto 2 39X</li>
</ol></li>
</ul></li>
</ul></li>
</ul></li>
</ul>
<p><img width="300" alt="" src="pictures/eDNAqPCR_protocolfile.jpeg"></p>
<ul>
<li>Run Setup:
<ul>
<li>Scan Mode: SYBR/FAM</li>
<li>Plate ID: I usually labeled with date and if it is a gblock plate or sample plate (eg. 2024-10-03_gBlock4)</li>
<li>Run File Name: I usually leave as the default (eg. BSB-eDNA_20241003_130438_795BR04561_gatins)</li>
<li>Save Location: either save to “my file” and extact after the run with a USB or have it upload to BR.io (what I usually do so that I can label each well while the machine is running)</li>
</ul></li>
</ul>
<p><img width="300" alt="" src="pictures/eDNAqPCR_protocolfile2.jpeg"></p>
<ul class="task-list">
<li><label><input type="checkbox">17. if uploading directly to BR.io, while the machine is running go to <a href="https://www.br.io">BR.io</a>, choose plate that is currently running and click on to plate setup to then label each well</label>
<ul>
<li>Fluorophore: FAM</li>
<li>Concentration: gblock concentration (eg. 10^8 = 1.00E+08)</li>
<li>Sample Type: standard</li>
<li>Sample Name: name of sample (eg. 10^8)</li>
</ul></li>
<li><label><input type="checkbox">18. before putting all concentration and samples back into freezer: make aloquots of 29 μl of the 10^6 copy number concentration to use as positive control on sample plates (freeze at -20 C in 0.5 ml tubes)</label></li>
<li><label><input type="checkbox">19. once plate is done, discard plate and look at results</label></li>
<li><label><input type="checkbox">20. export PCRD file and upload to CFX Maestro Software on lab laptop</label></li>
<li><label><input type="checkbox">21. exclude wells and concentrations where less than 95% of the replicates amplified to determine the limit of detection (usually low concentration such as 1 and 3)</label></li>
<li><label><input type="checkbox">22. check efficiency and R^2 values, if good, save slope and y-intercept numbers to use for later cq calculations</label>
<ul>
<li>efficiency should be between 90 and 105, ideally between 97 and 102</li>
<li>Slope should be between -3.3 and -3.5</li>
<li>R^2 should be between 0.995 and 1.00</li>
</ul></li>
</ul>
<p><img width="600" alt="" src="pictures/eDNAqPCR_gblockcurve.jpg"></p>
<p><strong>GBLOCK/STANDARD CURVE NOTES:</strong></p>
<ul>
<li>grayed out boxes = wells/replicates that have been excluded</li>
<li>Cq number = at which cycle each well amplified at/crosses the threshold line</li>
<li>concentration replicates should idealy be within 0.5 Cq cycles of each other</li>
<li>in example photo above, efficiency = 96.1% &amp; R^2 = 0.996</li>
<li>avoid freeze-thawing gBlock by making aloquots</li>
<li>keep everything on ice for as long as you can (including plate; plate ice packs can be found in -20 C freezer)</li>
<li>the probe is light sensetive</li>
<li>work in PCR hood when possible (cleaned with UV)</li>
</ul>
</section>
<section id="qpcr-for-samples" class="level2">
<h2 class="anchored" data-anchor-id="qpcr-for-samples">qPCR for samples</h2>
<p><strong>NEED:</strong></p>
<ul class="task-list">
<li><label><input type="checkbox">primer master mix</label>
<ul class="task-list">
<li><label><input type="checkbox">6.91 μl BSB-F</label></li>
<li><label><input type="checkbox">6.91 μl BSB-R</label></li>
<li><label><input type="checkbox">3.46 μl BSB-P</label></li>
<li><label><input type="checkbox">904.32 μl H20</label></li>
</ul></li>
<li><label><input type="checkbox"><a href="https://www.thermofisher.com/order/catalog/product/4396838">TaqMan™ Environmental Master Mix 2.0</a></label></li>
<li><label><input type="checkbox">ice</label></li>
<li><label><input type="checkbox">15 ml tube</label></li>
<li><label><input type="checkbox">1 or 2 ml microcentrifuge tube</label></li>
<li><label><input type="checkbox">PCR strip tubes</label></li>
<li><label><input type="checkbox">Hard-Shell PCR plate, 96-well, thin wall (<a href="https://www.bio-rad.com/en-us/sku/HSP9601-hard-shell-96-well-pcr-plates-low-profile-thin-wall-skirted-white-clear?ID=HSP9601">Bio-rad #HSP9601</a>; pkg of 50, white shell/clear well)</label></li>
<li><label><input type="checkbox">plate seal/cover (<a href="https://www.bio-rad.com/en-us/sku/17010701-microseal-b-pcr-plate-sealing-film-adhesive-optical?ID=17010701">Bio-rad Microseal ‘B’ seal Seals #MSB1001</a>)</label></li>
</ul>
<p><a href="https://docs.google.com/spreadsheets/d/1VQfYPV60tebJGguQfghsxpS0YRRm-WdmMyxA82zIlHk/edit?usp=sharing"><strong>PLATE PLAN:</strong></a></p>
<p><img width="900" alt="" src="pictures/eDNAqPCR_plateplan.png"></p>
<p><strong>PLATE NOTES:</strong></p>
<ul>
<li>each plate can hold up to 11 samples with 6 replicates each</li>
<li>each column is a new sample</li>
<li>the last two wells of each column is for positive control (i.e.&nbsp;spike)
<ul>
<li>spiked wells: 1 μl of 10^6 gblock &amp; 1 μl of sample or water (same as rest of column)</li>
<li>the purpose of the spiked wells is to check for DNA inhabition</li>
</ul></li>
<li>column 12 is for negative control; replace the 2 μl sample with 2 μl nuclease-free water</li>
<li>work in PCR hood as much as possible</li>
<li>be very careful with gBlock since it is very concentrated arificial DNA</li>
</ul>
<p><strong>PREP:</strong></p>
<ul class="task-list">
<li><label><input type="checkbox">1. wipe down PCR hood and counter with 10% bleach</label></li>
<li><label><input type="checkbox">2. sterilize hood with UV light</label></li>
<li><label><input type="checkbox">3. grab ice</label></li>
<li><label><input type="checkbox">4. thaw primer master mix or individual primers/probe (if you have to make more master mix) in ice container with lid to block light (to protect probe)</label></li>
<li><label><input type="checkbox">5. thaw 10^6 gBlock concentration aloquot in ice bucket</label></li>
<li><label><input type="checkbox">6. start to thaw samples</label></li>
<li><label><input type="checkbox">7. make and print plate plan</label></li>
<li><label><input type="checkbox">8. tape plate plan to hood</label></li>
</ul>
<p><strong>STEPS:</strong></p>
<ul class="task-list">
<li><p><label><input type="checkbox">1. pipet 1,152 μl of the taquman master mix into a 15 ml tube</label></p></li>
<li><p><label><input type="checkbox">2. then pipet 921.6 μl of the primer master mix into the same tube</label></p></li>
<li><p><label><input type="checkbox">3. swirl gently, and transfer half of the liquid to a 1.5 or 2 ml tube to make it easier to pipet from (transfer the remaining combined master mix once it runs low)</label></p></li>
<li><p><label><input type="checkbox">4. pipet 129.6 μl of the combined master mix into each well of a strip tube</label></p></li>
<li><p><label><input type="checkbox">5. using a multichannel pipet, pipet out 18 μl of strip tube solution into each well of the plate</label></p></li>
<li><p><label><input type="checkbox">6. after 6 columns, pipet out another 129.6 μl of the combined master mix into each well of the same strip tube (might need to pipet it around if running low in specific tubes of the strip tube)</label></p></li>
<li><p><label><input type="checkbox">7. Finish pipetting 18 μl of the strip tube solution into the remaining wells of the plate</label></p></li>
<li><p><label><input type="checkbox">8. starting with column 12 to avoid contaminating the neg control wells, pipet 2 μl of nuclease-free water into each well in rows A-F (as shown in the blank plate plan above)</label></p></li>
<li><p><label><input type="checkbox">9. pipet 1 μl of nuclease-free water into wells G12 &amp; H12</label></p></li>
<li><p><label><input type="checkbox">10. vortex and centrifuge first sample (sample going in column 11), before pipetting 2 μl in each well for rows A-F then 1 μl for rows G-H</label></p></li>
<li><p><label><input type="checkbox">11. repeat step 10 for every sample working right to left</label></p></li>
<li><p><label><input type="checkbox">12. vortex and centrifuge thawed gBlock aliquot and pipet 1 μl into every well in rows G-H (making sure to change tips for every sample)</label></p></li>
<li><p><label><input type="checkbox">13. place down plate seal, making sure it covers every well</label></p></li>
<li><p><label><input type="checkbox">14. vortex, then bring to the shared lab space to centrifuge for ~10-20 secounds by holding down the short spin button (you can find a plate balance in the lab next to the empty tube racks)</label></p></li>
<li><p><label><input type="checkbox">15. wipe down seal with a kimwipe</label></p></li>
<li><p><label><input type="checkbox">16. place into Bio-rad qPCR machine and start BSB eDNA run (labeled: BSB-eDNA) which can be found in “Files” under “My Files”</label></p>
<ul>
<li>BSB-eDNA Protocol File:
<ul>
<li>Modified: Aug 28, 2024, 2:13 PM</li>
<li>Method: CALC</li>
<li>Lid Temperature: 105 °C</li>
<li>Reaction Volume: 20 μl
<ul>
<li>Steps:
<ol type="1">
<li>95 °C, 00:10:00</li>
<li>95 °C, 00:00:10</li>
<li>60 °C, 00:00:45 Plate Read</li>
<li>Goto 2 39X</li>
</ol></li>
</ul></li>
</ul></li>
</ul></li>
</ul>
<p><img width="300" alt="" src="pictures/eDNAqPCR_protocolfile.jpeg"></p>
<ul>
<li>Run Setup:
<ul>
<li>Scan Mode: SYBR/FAM</li>
<li>Plate ID: I usually labeled with date and if it is a gblock plate or sample plate (eg. 2024-11-05_plate5)</li>
<li>Run File Name: I usually leave as the default (eg. 20241105_112821_CT023639_HOGAN.zpcr)</li>
<li>Save Location: either save to “my file” and extact after the run with a USB or have it upload to BR.io (what I usually do so that I can label each well while the machine is running)</li>
</ul></li>
</ul>
<p><img width="300" alt="" src="pictures/eDNAqPCR_protocolfile2.jpeg"></p>
<ul class="task-list">
<li><label><input type="checkbox">17. if uploading directly to BR.io, while the machine is running go to <a href="https://www.br.io">BR.io</a>, choose plate that is currently running and click on to plate setup to then label each well</label>
<ul>
<li>Fluorophore: FAM</li>
<li>Sample Type: unknown (columns 1-11, rows A-F), positive control (columns 1-12, rows G-H), negative control (column 12, rows A-F)</li>
<li>Sample Name: name of sample (eg. 5TW_6hr_2410_01)</li>
</ul></li>
<li><label><input type="checkbox">18. once plate is done, discard plate and look at results on BR.io</label></li>
<li><label><input type="checkbox">19. export PCRD file and upload to CFX Maestro Software on lab laptop</label></li>
</ul>
<p><strong>qPCR notes:</strong></p>
<ul>
<li>for unknown reasons (possibly due to contamination in the qPCR machine), there is a change that samples where very low or no amounts of DNA will amplify very early in the cycles but not increase the same as normal amplifications; if this occurs exclude any odd amplification replicate/well from analysis</li>
</ul>
<p>example of odd amplification vs normal amplification for one sample (in log scale):</p>
<p><img width="600" alt="" src="pictures/eDNAqPCR_oddsample.jpg"></p>
<ul>
<li>normal amplification should occur in cycles later than your highest gBlock concentrations (eg. 10^8 or 10^6)</li>
<li>positive detection is assumed for a samples as long as at least one replicate was positive</li>
</ul>
</section>
<section id="calculating-copy-number" class="level2">
<h2 class="anchored" data-anchor-id="calculating-copy-number">Calculating Copy Number</h2>
<p><strong>FORMULA:</strong></p>
<p>Copy number = 10^(([average Cq value from qPCR results] - [y-int])/ [slope])</p>
<p><strong>STEPS:</strong></p>
<ul class="task-list">
<li><label><input type="checkbox">1. open the PCRD file with the CFX Maestro Software on lab laptop</label></li>
</ul>
<p>it will look like this:</p>
<p><img width="600" alt="" src="pictures/eDNAqPCR_wells.png"></p>
<ul class="task-list">
<li><label><input type="checkbox">2. to label wells if it has not already been done = go to settings -&gt; plate setup -&gt; view/edit plate…</label></li>
<li><label><input type="checkbox">3. while also on this page, choose sample replicates so analysis is easier later on</label></li>
<li><label><input type="checkbox">4. check for any oddities by going through all replicates for one sample at a time (unhighlight all wells then highlight only wells you want to look at)</label>
<ul>
<li>4.1 see qPCR notes above for examples</li>
<li>4.2 all spike + sample wells should have amplified and been between +/- 2 of spike + nuclease-free water wells</li>
<li>4.3 negative control wells should have no amplification (cq = N/A)</li>
</ul></li>
</ul>
<p>example:</p>
<p><img width="600" alt="" src="pictures/eDNAqPCR_lesswells.png"></p>
<ul class="task-list">
<li><label><input type="checkbox">5. once done looking over all the wells, exluded wells as needed (by right clicking and choosing exclude from analysis)</label></li>
<li><label><input type="checkbox">6. rehighlight all the wells and export all the datasheets to excel from CFX Maestro Software</label></li>
<li><label><input type="checkbox">7. open the Quantification Cq Results excel sheet</label></li>
</ul>
<p>example Quantification Cq Results sheet (zoomed in; only showing replicates for ~2-3 samples):</p>
<p><img width="600" alt="" src="pictures/eDNAqPCR_results.jpg"></p>
<ul class="task-list">
<li><label><input type="checkbox">8. copy Cq Mean for each sample from Quantification Cq Results and paste into datasheet (make avg Cq column) or if did not group samples by replicate (see content column above), manually average Cq for each sample (usually 6 reps per sample) then paste into datasheet</label></li>
<li><label><input type="checkbox">9. input Cq value into copy number formula and use y-intercept and slope from standard curve/gblock plate</label></li>
<li><label><input type="checkbox">10. repeat for all samples</label></li>
</ul>


</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
      const outerScaffold = trigger.parentElement.cloneNode(true);
      const codeEl = outerScaffold.querySelector('code');
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp("https:\/\/remygatins\.github\.io\/GatinsLabProtocols\/");
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script>
</div> <!-- /content -->




</body></html>