<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.5.54">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>lab_molec_illuminalibraryprep – Gatins lab</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


<link rel="stylesheet" href="styles.css">
</head>

<body class="nav-sidebar docked nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="./index.html">
    <span class="navbar-title">Gatins lab</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link active" href="./index.html" aria-current="page"> 
<span class="menu-text">Welcome</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./lab_generalprotocols.html"> 
<span class="menu-text">Lab Protocols</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./fieldprotocols.html"> 
<span class="menu-text">Field Protocols</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./prof_development.html"> 
<span class="menu-text">Professional Development</span></a>
  </li>  
</ul>
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/remygatins"> <i class="bi bi-github" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://twitter.com/remygatins?lang=en"> <i class="bi bi-twitter" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
</div>
      </div> <!-- /container-fluid -->
    </nav>
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./lab_molec_dnaextraction.html">Molecular Protocols</a></li><li class="breadcrumb-item"><a href="./lab_molec_illuminalibraryprep.html">Illumina Library Prep</a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation docked overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header sidebar-header-stacked">
      <a href="./index.html" class="sidebar-logo-link">
      <img src="./pictures/Gatins_Lab_logo_dark_simple.png" alt="" class="sidebar-logo py-0 d-lg-inline d-none">
      </a>
      </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="false">
 <span class="menu-text">General Protocols</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./lab_generalprotocols.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">General Protocols</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./lab_gen_stableisotope.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Stable Isotope Protocol</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="false">
 <span class="menu-text">Equipment</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./lab_equip_autoclave.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Autoclave (OGL)</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./lab_equip_tapestation.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Tapestation</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./lab_equip_qubit.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">QubitFlex Fluorometer</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="true">
 <span class="menu-text">Molecular Protocols</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./lab_molec_dnaextraction.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Qiagen DNeasy extraction</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./lab_molec_DNA_beadcleanup.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">DNA Bead Cleanup</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./lab_molec_illuminalibraryprep.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">Illumina Library Prep</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./lab_molec_adapter_concentration.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Adapter Concentrations</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./low_te_buffer.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Protocol for making 20x Low TE Buffer</span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#illumina-library-prep" id="toc-illumina-library-prep" class="nav-link active" data-scroll-target="#illumina-library-prep">Illumina Library Prep</a>
  <ul class="collapse">
  <li><a href="#kapa-hyperplus-kit" id="toc-kapa-hyperplus-kit" class="nav-link" data-scroll-target="#kapa-hyperplus-kit">KAPA HyperPlus kit</a></li>
  <li><a href="#prepare-dna" id="toc-prepare-dna" class="nav-link" data-scroll-target="#prepare-dna">0. Prepare DNA</a></li>
  <li><a href="#fragmentation" id="toc-fragmentation" class="nav-link" data-scroll-target="#fragmentation">1. Fragmentation</a></li>
  <li><a href="#end-repair-and-a-tailing" id="toc-end-repair-and-a-tailing" class="nav-link" data-scroll-target="#end-repair-and-a-tailing">2. End Repair and A-tailing</a></li>
  <li><a href="#adapter-ligation" id="toc-adapter-ligation" class="nav-link" data-scroll-target="#adapter-ligation">3. Adapter Ligation</a></li>
  <li><a href="#post-ligation-cleanup" id="toc-post-ligation-cleanup" class="nav-link" data-scroll-target="#post-ligation-cleanup">4. Post Ligation Cleanup</a></li>
  <li><a href="#library-amplification" id="toc-library-amplification" class="nav-link" data-scroll-target="#library-amplification">5. Library Amplification</a></li>
  <li><a href="#post-ligation-cleanup-1" id="toc-post-ligation-cleanup-1" class="nav-link" data-scroll-target="#post-ligation-cleanup-1">6. Post-ligation Cleanup</a></li>
  <li><a href="#size-selection" id="toc-size-selection" class="nav-link" data-scroll-target="#size-selection">7. Size Selection</a>
  <ul class="collapse">
  <li><a href="#a-uppercut" id="toc-a-uppercut" class="nav-link" data-scroll-target="#a-uppercut">7.a Uppercut</a></li>
  <li><a href="#b-lowercut" id="toc-b-lowercut" class="nav-link" data-scroll-target="#b-lowercut">7.b Lowercut</a></li>
  </ul></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">




<section id="illumina-library-prep" class="level1">
<h1>Illumina Library Prep</h1>
<section id="kapa-hyperplus-kit" class="level2">
<h2 class="anchored" data-anchor-id="kapa-hyperplus-kit">KAPA HyperPlus kit</h2>
<p>Kit KR 1145</p>
<p><a href="https://github.com/remygatins/GatinsLab/edit/main/lab_molec_illuminalibraryprep.qmd">Edit this page</a></p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><embed src="https://rochesequencingstore.com/wp-content/uploads/2017/10/KAPA-HyperPlus-Kit_KR1145-%E2%80%93-v5.19.pdf" class="img-fluid"></p>
<figcaption>Manual</figcaption>
</figure>
</div>
</section>
<section id="prepare-dna" class="level2">
<h2 class="anchored" data-anchor-id="prepare-dna">0. Prepare DNA</h2>
<p>Removal of EDTA from DNA samples prior to fragmentation is recommended to ensure consistent results. This may be achieved by means of a 3X bead based cleanup with KAPA cleanup beads. Please refer to the relevant Technical Data Sheet (KR1705 or KR1245) for a detailed DNA cleanup protocol.</p>
<p>For optimal fragmentation results, <strong>elute DNA in Elution Buffer (10 mM Tris-HCl, pH 8.0 - 8.5)</strong> after the cleanup.</p>
<p><a href="https://rochesequencingstore.com/wp-content/uploads/2022/07/KAPA-Pure-Beads-Technical-Data-Sheet.pdf">KAPA cleanup bead protocol</a></p>
</section>
<section id="fragmentation" class="level2">
<h2 class="anchored" data-anchor-id="fragmentation">1. Fragmentation</h2>
<ul class="task-list">
<li><label><input type="checkbox">1.1 Dilute 250 ng of dsDNA in Elution Buffer (10 mM Tris-HCl, pH 8.0 - 8.5) to reach a total volume of 11.7 µL (1/3 rx)</label>
<ul>
<li>only do this if samples do not contain EDTA. Otherwise see original protocol for instructions to clean DNA or use Conditioning Solution.</li>
</ul></li>
<li><label><input type="checkbox">1.2 Mix by gentle vortexing or pipetting up and down</label></li>
<li><label><input type="checkbox">1.3 Assemble the fragmentation reaction on ice by adding the following components in order:</label></li>
</ul>
<table class="caption-top table">
<thead>
<tr class="header">
<th style="text-align: left;">Component</th>
<th style="text-align: left;">Volume</th>
<th style="text-align: left;">1/3 Volume</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">250ng DNA</td>
<td style="text-align: left;">35 µL</td>
<td style="text-align: left;">11.7 µL</td>
</tr>
<tr class="even">
<td style="text-align: left;">KAPA Frag Buffer(10X)*</td>
<td style="text-align: left;">5 µL</td>
<td style="text-align: left;">1.6µL</td>
</tr>
<tr class="odd">
<td style="text-align: left;">KAPA Frag Enzyme*</td>
<td style="text-align: left;">10 µL</td>
<td style="text-align: left;">3.4 µL</td>
</tr>
<tr class="even">
<td style="text-align: left;">———————-</td>
<td style="text-align: left;">———</td>
<td style="text-align: left;">———–</td>
</tr>
<tr class="odd">
<td style="text-align: left;"><strong>Total Volume</strong></td>
<td style="text-align: left;"><strong>50 µL</strong></td>
<td style="text-align: left;"><strong>16.7 µL</strong></td>
</tr>
</tbody>
</table>
<p><em><strong>OPTIONAL</strong>: The KAPA Frag Buffer and Enzyme may be pre-mixed and kept on ice prior to reaction setup, and dispensed as a single solution</em>.</p>
<ul class="task-list">
<li><p><label><input type="checkbox"><strong>PREP</strong>: get the End Repair &amp; A-Tailing Buffer out of the freezer to begin melting for next step.</label></p></li>
<li><p><label><input type="checkbox">1.4 Vortex gently and spin down briefly. Return the plate/tue to ice and proceed immediately to the next step.</label></p></li>
<li><p><label><input type="checkbox">1.5 Incubate in a thermocycler, pre-cooled to 4°C and programmed as outlined below. A heated lid is not required for this step. If used, set the temperature of the heated lid to ≤50°C.</label></p></li>
</ul>
<table class="caption-top table">
<thead>
<tr class="header">
<th>Step</th>
<th>Temp</th>
<th>Time</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Pre-cool block</td>
<td>4°C</td>
<td>N/A</td>
</tr>
<tr class="even">
<td>Fragmentation</td>
<td>37°C</td>
<td>7:00*</td>
</tr>
<tr class="odd">
<td>HOLD</td>
<td>4°C</td>
<td>∞</td>
</tr>
</tbody>
</table>
<p>Fragmentation time was previously optimized for 500bp fragments starting from the optimization table below</p>
<table class="caption-top table">
<thead>
<tr class="header">
<th>Mode Fragment Length</th>
<th>Incubation time at 37°C</th>
<th>Optimization range</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>600 bp</td>
<td>5 min</td>
<td>3- 10 min</td>
</tr>
<tr class="even">
<td>350 bp</td>
<td>10 min</td>
<td>5- 20 min</td>
</tr>
<tr class="odd">
<td>200 bp</td>
<td>20 min</td>
<td>10- 25 min</td>
</tr>
<tr class="even">
<td>150 bp</td>
<td>30 min</td>
<td>20- 40 min</td>
</tr>
</tbody>
</table>
<ul class="task-list">
<li><label><input type="checkbox">1.6 Transfer reactions to ice, and proceed immediately to End Repair and A-tailing (step 2).</label></li>
</ul>
</section>
<section id="end-repair-and-a-tailing" class="level2">
<h2 class="anchored" data-anchor-id="end-repair-and-a-tailing">2. End Repair and A-tailing</h2>
<ul class="task-list">
<li><label><input type="checkbox">2.1 In the same plate/tube(s) in which enzymatic fragmentation was performed, assemble each End Repair and A-tailing reaction as follows:</label></li>
</ul>
<table class="caption-top table">
<thead>
<tr class="header">
<th>Component</th>
<th>Volume</th>
<th>1/3 Volume</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Fragmented, double-stranded DNA</td>
<td>50 µL</td>
<td>16.7 µL</td>
</tr>
<tr class="even">
<td>End Repair &amp; A-Tailing Buffer*</td>
<td>7 µL</td>
<td>2.43 µL</td>
</tr>
<tr class="odd">
<td>HyperPlus ERAT Enzyme Mix*</td>
<td>3 µL</td>
<td>1 µL</td>
</tr>
<tr class="even">
<td>——————–</td>
<td>——</td>
<td>——-</td>
</tr>
<tr class="odd">
<td><strong>Total Volume</strong></td>
<td><strong>60 µL</strong></td>
<td><strong>20 µL</strong></td>
</tr>
</tbody>
</table>
<p><em><strong>OPTIONAL</strong>: Pre-mix End Repair &amp; A-Tailing Buffer and HyperPlus ERAT Enzyme Mix</em> and keep on ice prior to reaction setup, and dispense as a single solution.</p>
<ul class="task-list">
<li><p><label><input type="checkbox">2.2 Vortex gently and spin down briefly. Return the reaction plate/tube(s) to ice. Proceed immediately to the next step.</label></p></li>
<li><p><label><input type="checkbox">2.3 Incubate in a thermocycler programmed as outlined below. A heated lid is required for this step. If possible, set the temperature of the heated lid to ~85°C (instead of the usual 105°C).</label></p></li>
</ul>
<table class="caption-top table">
<thead>
<tr class="header">
<th>Step</th>
<th>Temp</th>
<th>Time</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>End repair and A-tailing</td>
<td>65°C</td>
<td>30 min</td>
</tr>
<tr class="even">
<td>HOLD</td>
<td>4°C</td>
<td>∞</td>
</tr>
</tbody>
</table>
<ul class="task-list">
<li><p><label><input type="checkbox"><strong>PREP</strong>: Thaw reagents of the adapter ligation and pre-make the master mix.</label></p></li>
<li><p><label><input type="checkbox">2.4 Proceed immediately to Adapter Ligation (step 3).</label></p></li>
</ul>
</section>
<section id="adapter-ligation" class="level2">
<h2 class="anchored" data-anchor-id="adapter-ligation">3. Adapter Ligation</h2>
<ul class="task-list">
<li><label><input type="checkbox">3.1 Dilute adapter stocks to the appropriate concentration, as outlined in Table 4 (p.&nbsp;5) of the manual. <img width="600" alt="center" src="pictures/KapaHyperplus_Table4_Adapters.png"></label></li>
</ul>
<p>We added 250ng of DNA so we want an Adapter:insert molar ratio of 40:1. Since we are using 1/3 reaction volumes we combined the water and adapter stock volume to add 5 µL and calculated that we need a concentration of [6 µM]. For a more detailed explanation on this see the <a href="./lab_molec_adapter_concentration.html">Adapter Concentration Calculations</a> page.</p>
<ul class="task-list">
<li><label><input type="checkbox">3.2 In the same plate/tube(s) in which end repair andA-tailing was performed, assemble each adapter ligation reaction as follows:</label></li>
</ul>
<table class="caption-top table">
<thead>
<tr class="header">
<th>Component</th>
<th>Volume</th>
<th>1/3 Volume</th>
<th>Notes</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>End repair and A-tailing reaction product</td>
<td>60 µL</td>
<td>20 µL</td>
<td></td>
</tr>
<tr class="even">
<td>Adapter stock [6 µM]</td>
<td>5 µL</td>
<td>5 µL</td>
<td></td>
</tr>
<tr class="odd">
<td>PCR grade water*</td>
<td>5 µL</td>
<td>- µL</td>
<td>Already included in the adapter stock</td>
</tr>
<tr class="even">
<td>Ligation Buffer*</td>
<td>30 µL</td>
<td>10 µL</td>
<td></td>
</tr>
<tr class="odd">
<td>DNA Ligase*</td>
<td>10 µL</td>
<td>3.3 µL</td>
<td></td>
</tr>
<tr class="even">
<td>——————–</td>
<td>——</td>
<td>——</td>
<td>——</td>
</tr>
<tr class="odd">
<td><strong>Total Volume</strong></td>
<td><strong>110 µL</strong></td>
<td><strong>38.3 µL</strong></td>
<td></td>
</tr>
</tbody>
</table>
<p><strong>OPTIONAL</strong>: *The water, buffer and ligase enzyme should preferably be premixed and added in a single pipetting step. Premixes are stable for ≤24 hrs at room temperature, for ≤3 days at 2°C to 8°C, and for ≤4 weeks at -15°C to -25°C.</p>
<ul class="task-list">
<li><p><label><input type="checkbox"><strong>PREP</strong>: Acclimate Kapa beads to RT and make 80% ethanol for the bead cleanup</label></p></li>
<li><p><label><input type="checkbox">3.3 Mix thoroughly and centrifuge briefly.</label></p></li>
<li><p><label><input type="checkbox">3.4 Incubate at 20°C for 15 min.</label></p></li>
<li><p><label><input type="checkbox">3.5 Proceed immediately to Post-ligation Cleanup (step 4).</label></p></li>
</ul>
</section>
<section id="post-ligation-cleanup" class="level2">
<h2 class="anchored" data-anchor-id="post-ligation-cleanup">4. Post Ligation Cleanup</h2>
<p><strong>IMPORTANT</strong>- Kapa beads should be at RT and well mixed using the vortexer</p>
<ul class="task-list">
<li><p><label><input type="checkbox"><strong>PREP</strong>: Thaw KAPA HiFi HotStart ReadyMix (2X) of the next step.</label></p></li>
<li><p><label><input type="checkbox"><strong>PREP</strong>: Warm elution buffer to 37°C</label></p></li>
<li><p><label><input type="checkbox">4.1 In the same plate/tube(s), perform a 0.8X beadbased cleanup by combining the following:</label></p></li>
</ul>
<table class="caption-top table">
<thead>
<tr class="header">
<th>Component</th>
<th>Volume</th>
<th>1/3 Volume</th>
<th>Notes</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Adapter ligation reaction product</td>
<td>110 µL</td>
<td>38.3 µL</td>
<td></td>
</tr>
<tr class="even">
<td>KAPA cleanup beads (0.8X)</td>
<td>88 µL</td>
<td>30.64 µL</td>
<td></td>
</tr>
<tr class="odd">
<td>——————–</td>
<td>——</td>
<td>——</td>
<td>——</td>
</tr>
<tr class="even">
<td><strong>Total Volume</strong></td>
<td><strong>198 µL</strong></td>
<td><strong>68.9 µL</strong></td>
<td></td>
</tr>
</tbody>
</table>
<ul class="task-list">
<li><p><label><input type="checkbox">4.2 Mix thoroughly by vortexing and/or pipetting up and down multiple times.</label></p></li>
<li><p><label><input type="checkbox">4.3 Incubate the plate/tube(s) at room temperature for 15 min or less to bind DNA to the beads.</label></p></li>
<li><p><label><input type="checkbox">4.4 Place the plate/tube(s) on a magnet to capture the beads. Incubate until the liquid is clear.</label></p></li>
<li><p><label><input type="checkbox">4.5 Carefully remove and discard the supernatant.</label></p></li>
<li><p><label><input type="checkbox">4.6 Keeping the plate/tube(s) on the magnet, add 100 µL of 80% ethanol.</label></p></li>
<li><p><label><input type="checkbox">4.7 Incubate the plate/tube(s) on the magnet at room temperature for ≥30 sec.</label></p></li>
<li><p><label><input type="checkbox">4.8 Carefully remove and discard the ethanol.</label></p></li>
<li><p><label><input type="checkbox">4.9 Keeping the plate/tube(s) on the magnet, add 100 µL of 80% ethanol.</label></p></li>
<li><p><label><input type="checkbox">4.10 Incubate the plate/tube(s) on the magnet at room temperature for ≥30 sec.</label></p></li>
<li><p><label><input type="checkbox">4.11 Carefully remove and discard the ethanol. Try to remove all residual ethanol without disturbing the beads.</label></p></li>
<li><p><label><input type="checkbox">4.12 Dry the beads at room temperature for 3 – 5 min, or until all of the ethanol has evaporated. <strong>Caution: over drying the beads may result in reduced yield.</strong></label></p></li>
<li><p><label><input type="checkbox">4.13 Remove the plate/tube(s) from the magnet.</label></p></li>
<li><p><label><input type="checkbox">4.14 Resuspend the beads in 25 µL of elution buffer (10 mM Tris-HCl, pH 8.0 – 8.5) previously warmed to 37°C</label></p></li>
<li><p><label><input type="checkbox">4.15 Incubate the plate/tube(s) at room temperature for 5 min to elute DNA off the beads.</label></p></li>
<li><p><label><input type="checkbox">4.16 Place the plate/tube(s) on a magnet to capture the beads. Incubate until the liquid is clear.</label></p></li>
<li><p><label><input type="checkbox">4.17 Transfer 20 µL of supernatant to a new plate/tube(s) to proceed with Library Amplification (step 5)</label></p></li>
</ul>
</section>
<section id="library-amplification" class="level2">
<h2 class="anchored" data-anchor-id="library-amplification">5. Library Amplification</h2>
<p>The KAPA Library Amplification Primer Mix (10X) is designed to eliminate or delay primer depletion during library amplification reactions performed with KAPA HiFi HotStart ReadyMix. The primer mix is suitable for the amplification of all Illumina libraries flanked by the P5 and P7 flow cell sequences. Primers are supplied at a 10X concentration of 20 μM each, and have been formulated as described below. User-supplied primer mixes may be used in combination with incomplete or custom adapters. Please contact Technical Support at sequencing.roche.com/support for guidelines on the formulation of user-supplied library amplification primers.</p>
<p>• To achieve the highest amplification efficiency and avoid primer depletion, it is critical to use an optimal concentration of high-quality primers. Primers should be used at a final concentration of 0.5 – 4 μM each. For libraries constructed from ≥100 ng input DNA, a final concentration of at least 2 μM of each primer is recommended.</p>
<p>• Library amplification primers should be HPLC-purified and modified to include a phosphorothioate bond at the 3’-terminal of each primer (to prevent degradation by the strong proofreading activity of KAPA HiFi HotStart). Always store and dilute primers in a buffered solution (e.g., 10mM Tris-HCl, pH8.0–8.5), and limit the number of freeze-thaw cycles. To achieve the latter, store primers at 2°C to 8°C for short-term use, or as single-use aliquots at -15°C to -25°C.</p>
<ul class="task-list">
<li><label><input type="checkbox">5.1 Assemble each library amplification reaction as follows:</label></li>
</ul>
<table class="caption-top table">
<colgroup>
<col style="width: 31%">
<col style="width: 22%">
<col style="width: 31%">
<col style="width: 14%">
</colgroup>
<thead>
<tr class="header">
<th>Component</th>
<th>Volume</th>
<th>1/3 Volume</th>
<th>Notes</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>KAPA HiFi HotStart ReadyMix (2X)</td>
<td>25 µL</td>
<td>8.3 µL</td>
<td></td>
</tr>
<tr class="even">
<td>Library Amplification Primer Mix (10X)*</td>
<td>5 µL</td>
<td>1.7 µL</td>
<td>if using custon Primers i5/i7 they should be at [20µM]</td>
</tr>
<tr class="odd">
<td>Adapter-ligated library</td>
<td>20 µL</td>
<td>6.7 µL</td>
<td></td>
</tr>
<tr class="even">
<td>——————–</td>
<td>——</td>
<td>——</td>
<td>——</td>
</tr>
<tr class="odd">
<td><strong>Total Volume</strong></td>
<td><strong>50 µL</strong></td>
<td><strong>16.7 µL</strong></td>
<td></td>
</tr>
</tbody>
</table>
<p>** NOTE: keep Adapter ligated library extra just in case **</p>
<ul class="task-list">
<li><label><input type="checkbox">5.2 Mix thoroughly and centrifuge briefly.</label></li>
<li><label><input type="checkbox">5.3 Amplify using the following cycling protocol:</label></li>
</ul>
<table class="caption-top table">
<thead>
<tr class="header">
<th>Step</th>
<th>Temp</th>
<th>Time</th>
<th>Cycles</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Initial Denaturation</td>
<td>98°C</td>
<td>45 sec</td>
<td>1</td>
</tr>
<tr class="even">
<td>Denaturation</td>
<td>98°C</td>
<td>15 sec</td>
<td></td>
</tr>
<tr class="odd">
<td>Annealing</td>
<td>60°C</td>
<td>30 sec</td>
<td>8X</td>
</tr>
<tr class="even">
<td>Extension</td>
<td>72°C</td>
<td>30 sec</td>
<td></td>
</tr>
<tr class="odd">
<td>Final Extension</td>
<td>72°C</td>
<td>1 min</td>
<td>1</td>
</tr>
<tr class="even">
<td>HOLD</td>
<td>4°C</td>
<td>∞</td>
<td>1</td>
</tr>
</tbody>
</table>
<ul class="task-list">
<li><label><input type="checkbox">5.4 Proceed directly to Post-amplification Cleanup (step 6).</label></li>
</ul>
</section>
<section id="post-ligation-cleanup-1" class="level2">
<h2 class="anchored" data-anchor-id="post-ligation-cleanup-1">6. Post-ligation Cleanup</h2>
<p><strong>IMPORTANT- Only do this the first time and check your library. Otherwise we will skip and go directly to the size selection step.</strong></p>
<ul class="task-list">
<li><label><input type="checkbox">6.1 In the same plate/tube(s), perform a 1X beadbased cleanup by combining the following:</label></li>
</ul>
<table class="caption-top table">
<thead>
<tr class="header">
<th>Component</th>
<th>Volume</th>
<th>1/3 Volume</th>
<th>Notes</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Library amplification reaction product</td>
<td>50 µL</td>
<td>16.7 µL</td>
<td></td>
</tr>
<tr class="even">
<td>KAPA cleanup beads (1X)</td>
<td>50 µL</td>
<td>16.7 µL</td>
<td></td>
</tr>
<tr class="odd">
<td>——————–</td>
<td>——</td>
<td>——</td>
<td>——</td>
</tr>
<tr class="even">
<td><strong>Total Volume</strong></td>
<td><strong>100 µL</strong></td>
<td><strong>33.4 µL</strong></td>
<td></td>
</tr>
</tbody>
</table>
<ul class="task-list">
<li><p><label><input type="checkbox">6.2 Mix thoroughly by vortexing and/or pipetting up and down multiple times.</label></p></li>
<li><p><label><input type="checkbox">6.3 Incubate the plate/tube(s) at room temperature for 15 min or less to bind DNA to the beads.</label></p></li>
<li><p><label><input type="checkbox">6.4 Place the plate/tube(s) on a magnet to capture the beads. Incubate until the liquid is clear.</label></p></li>
<li><p><label><input type="checkbox">6.5 Carefully remove and discard the supernatant.</label></p></li>
<li><p><label><input type="checkbox">6.6 Keeping the plate/tube(s) on the magnet, add 100 µL of 80% ethanol.</label></p></li>
<li><p><label><input type="checkbox">6.7 Incubate the plate/tube(s) on the magnet at room temperature for ≥30 sec.</label></p></li>
<li><p><label><input type="checkbox">6.8 Carefully remove and discard the ethanol.</label></p></li>
<li><p><label><input type="checkbox">6.9 Keeping the plate/tube(s) on the magnet, add 100 µL of 80% ethanol.</label></p></li>
<li><p><label><input type="checkbox">6.10 Incubate the plate/tube(s) on the magnet at room temperature for ≥30 sec.</label></p></li>
<li><p><label><input type="checkbox">6.11 Carefully remove and discard the ethanol. Try to remove all residual ethanol without disturbing the beads.</label></p></li>
<li><p><label><input type="checkbox">6.12 Dry the beads at room temperature for 3 – 5 min, or until all of the ethanol has evaporated. <strong>Caution: over drying the beads may result in reduced yield.</strong></label></p></li>
<li><p><label><input type="checkbox">6.13 Remove the plate/tube(s) from the magnet.</label></p></li>
<li><p><label><input type="checkbox">6.14 Resuspend the beads in 25 µL of elution buffer (10 mM Tris-HCl, pH 8.0 – 8.5) previously warmed to 37°C</label></p></li>
<li><p><label><input type="checkbox">6.15 Incubate the plate/tube(s) at room temperature for 5 min to elute DNA off the beads.</label></p></li>
<li><p><label><input type="checkbox">6.16 Place the plate/tube(s) on a magnet to capture the beads. Incubate until the liquid is clear.</label></p></li>
<li><p><label><input type="checkbox">6.17 Transfer clear supernatant to a new plate/tube(s) and proceed with size selection (step 7)</label></p></li>
</ul>
</section>
<section id="size-selection" class="level2">
<h2 class="anchored" data-anchor-id="size-selection">7. Size Selection</h2>
<p>*The original protocol calls for a size selection using 0.7X and 0.9X size selection (~ 250bp - 400bp). We have tested</p>
<section id="a-uppercut" class="level3">
<h3 class="anchored" data-anchor-id="a-uppercut">7.a Uppercut</h3>
<p><em>We are first going to remove any fragments larger than &gt; 600 bp</em></p>
<ul class="task-list">
<li><p><label><input type="checkbox">7.a.1 First, add 13.3 µL of molecular H2O to increase the final volume after the PCR from 16.7 µL to 30 µL.</label></p></li>
<li><p><label><input type="checkbox">7.a.2 In the same plate/tube(s), perform a <strong>0.56X</strong> beadbased cleanup by combining the following:</label></p></li>
</ul>
<table class="caption-top table">
<thead>
<tr class="header">
<th>Component</th>
<th>Volume</th>
<th>1/3 Volume</th>
<th>Notes</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Library amplification reaction product + H20</td>
<td>50 µL</td>
<td>30 µL</td>
<td></td>
</tr>
<tr class="even">
<td>KAPA cleanup beads <strong>(0.56X)</strong></td>
<td>28 µL</td>
<td>16.8 µL</td>
<td></td>
</tr>
<tr class="odd">
<td>——————–</td>
<td>——</td>
<td>——</td>
<td>——</td>
</tr>
<tr class="even">
<td><strong>Total Volume</strong></td>
<td><strong>78 µL</strong></td>
<td><strong>46.8 µL</strong></td>
<td></td>
</tr>
</tbody>
</table>
<ul class="task-list">
<li><p><label><input type="checkbox">7.a.3 Mix thoroughly by vortexing and/or pipetting up and down multiple times.</label></p></li>
<li><p><label><input type="checkbox">7.a.4 Incubate the plate/tube(s) at room temperature for 15 min or less to bind DNA to the beads.</label></p></li>
<li><p><label><input type="checkbox">7.a.5 Place the plate/tube(s) on a magnet to capture the beads. Incubate until the liquid is clear.</label></p></li>
<li><p><label><input type="checkbox">7.a.6 Carefully remove and <strong>KEEP THE SUPERNATANT and discard the beads</strong></label></p></li>
</ul>
</section>
<section id="b-lowercut" class="level3">
<h3 class="anchored" data-anchor-id="b-lowercut">7.b Lowercut</h3>
<p><em>We are now going to retain fragments &gt; ~300 bp</em></p>
<ul class="task-list">
<li><label><input type="checkbox">7.b.1 In the same plate/tube(s), perform a <strong>0.72X</strong> beadbased cleanup by combining the following:</label></li>
</ul>
<table class="caption-top table">
<colgroup>
<col style="width: 31%">
<col style="width: 22%">
<col style="width: 31%">
<col style="width: 14%">
</colgroup>
<thead>
<tr class="header">
<th>Component</th>
<th>Volume</th>
<th>1/3 Volume</th>
<th>Notes</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Uppercut supernatant (Amplification product)</td>
<td>50 µL</td>
<td>46.8 µL</td>
<td></td>
</tr>
<tr class="even">
<td>KAPA cleanup beads (<strong>0.72X</strong>)</td>
<td>36 µL</td>
<td>21.6 µL</td>
<td>calculated from previous volume of 30 µL</td>
</tr>
<tr class="odd">
<td>——————–</td>
<td>——</td>
<td>——</td>
<td>——</td>
</tr>
<tr class="even">
<td><strong>Total Volume</strong></td>
<td><strong>86 µL</strong></td>
<td><strong>68.4 µL</strong></td>
<td></td>
</tr>
</tbody>
</table>
<ul class="task-list">
<li><p><label><input type="checkbox">7.b.2 Mix thoroughly by vortexing and/or pipetting up and down multiple times.</label></p></li>
<li><p><label><input type="checkbox">7.b.3 Incubate the plate/tube(s) at room temperature for 15 min or less to bind DNA to the beads.</label></p></li>
<li><p><label><input type="checkbox">7.b.4 Place the plate/tube(s) on a magnet to capture the beads. Incubate until the liquid is clear.</label></p></li>
<li><p><label><input type="checkbox">7.b.5 Carefully remove and discard the supernatant.</label></p></li>
<li><p><label><input type="checkbox">7.b.6 Keeping the plate/tube(s) on the magnet, add 100 µL of 80% ethanol.</label></p></li>
<li><p><label><input type="checkbox">7.b.7 Incubate the plate/tube(s) on the magnet at room temperature for ≥30 sec.</label></p></li>
<li><p><label><input type="checkbox">7.b.8 Carefully remove and discard the ethanol.</label></p></li>
<li><p><label><input type="checkbox">7.b.9 Keeping the plate/tube(s) on the magnet, add 100 µL of 80% ethanol.</label></p></li>
<li><p><label><input type="checkbox">7.b.10 Incubate the plate/tube(s) on the magnet at room temperature for ≥30 sec.</label></p></li>
<li><p><label><input type="checkbox">7.b.11 Carefully remove and discard the ethanol. Try to remove all residual ethanol without disturbing the beads.</label></p></li>
<li><p><label><input type="checkbox">7.b.12 Dry the beads at room temperature for 3 – 5 min, or until all of the ethanol has evaporated. <strong>Caution: over drying the beads may result in reduced yield.</strong></label></p></li>
<li><p><label><input type="checkbox">7.b.13 Remove the plate/tube(s) from the magnet.</label></p></li>
<li><p><label><input type="checkbox">7.b.14 Resuspend the beads in 22 µL of elution buffer (10 mM Tris-HCl, pH 8.0 – 8.5) previously warmed to 37°C</label></p></li>
<li><p><label><input type="checkbox">7.b.15 Incubate the plate/tube(s) at room temperature for 5 min to elute DNA off the beads.</label></p></li>
<li><p><label><input type="checkbox">7.b.16 Place the plate/tube(s) on a magnet to capture the beads. Incubate until the liquid is clear.</label></p></li>
<li><p><label><input type="checkbox">7.b.17 Transfer 20 µL of clear supernatant (Amplified Library) to a new plate/tube(s)</label></p></li>
</ul>
<p>Congratulations! You are done! Now run your QC steps (e.g., Tapestation, Qubit, RT-PCR) prior to sending for sequencing.</p>


</section>
</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp("https:\/\/remygatins\.github\.io\/GatinsLab\/");
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>